File: 1759.cpp
#include <stdio.h>
#include <algorithm>
#include <iostream>
#include <string>
#include <vector>

using namespace std;

char vowel[5] = {'a', 'e', 'i', 'o', 'u'};
int code_len = 0;

bool check_condition(std::string code){
  int count = 0;
  for (int i = 0; i < 5; i++){
    count += (code.find(vowel[i]) != -1);
  }

  if (count >= 1 && code.size() - count >= 2){
    return true;
  }

  else{
    return false;
  }
}

void get_ans(std::vector<char> letter, int index, string code){

  if (code.size() == code_len && check_condition(code)){
    cout << code << endl;
    return;
  }
  else if (code.size() < code_len){
    string origin = code;
    int remain_len = code_len - code.size();
    for (int i = index; i < letter.size() - remain_len + 1; i++){
      string tmp_code = origin + letter[i];
      get_ans(letter, i + 1, tmp_code);
    }
  }
}





int main(){
  int letter_len = 0;
  scanf("%d", &code_len);
  scanf("%d", &letter_len);

  std::vector<char> letter(letter_len);

  for (int i = 0; i < letter_len; i++)
  {
    getchar();
    scanf("%c", &letter[i]);

  }

  sort(letter.begin(), letter.end());


  get_ans(letter, 0, "");

}

File: 1759.py
import sys

ans = []
code_length = 0

def check_condition(code):
    count1 = code.count('a') + code.count('e') + code.count('i') + code.count('o') + code.count('u')
    count2 = len(code) - count1
    if count1 >= 1 and count2 >= 2:
        return True
    else:
        return False

def get_ans(letter, pos, code):

    if len(code) == code_length and check_condition(code):
        ans.append(code) 
        return
    elif len(code) < code_length:
        origin = code
        remain_length = code_length - len(code)
        for i in range(pos, len(letter) - remain_length + 1):
            tmp_code = origin + letter[i]
            get_ans(letter, i + 1, tmp_code)




code_length, letter_length = map(int, sys.stdin.readline().split())

letter = list(sys.stdin.readline().split())

letter.sort()

get_ans(letter, 0, '')

print('\n'.join(ans))

